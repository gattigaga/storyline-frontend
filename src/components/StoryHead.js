import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

import Avatar from "./Avatar";
import Button from "./Button";

const Container = styled.div`
  display: flex;
  margin-bottom: 32px;
`;

const Info = styled.div`
  flex: 1;
  margin-left: 16px;
`;

const Name = styled(Link)`
  font-size: 12px;
  font-family: Roboto;
  font-weight: bold;
  color: #333;
  margin: 0px;
  margin-right: 8px;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;

const Description = styled.p`
  font-size: 12px;
  font-family: Roboto;
  font-weight: normal;
  color: #aaa;
  margin: 0px;
  margin-bottom: 12px;
`;

const Date = Description.extend`
  color: #e74c3c;
  font-weight: normal;
  margin: 0px;
`;

const Row = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 4px;
`;

export const MiniButton = styled(Button)`
  padding: 2px 12px;
  font-size: 10px;
`;

const StoryHead = ({
  className,
  avatar,
  name,
  description,
  date,
  onClickFollow,
  isFollowed,
  href
}) => (
  <Container className={className}>
    <Link to={href}>
      <Avatar src={avatar} />
    </Link>
    <Info>
      <Row>
        <Name to={href}>{name}</Name>
        <MiniButton
          caption={isFollowed ? "Unfollow" : "Follow"}
          onClick={onClickFollow}
          isOutlined={!isFollowed}
        />
      </Row>
      <Description>{description}</Description>
      <Date>{date}</Date>
    </Info>
  </Container>
);

StoryHead.propTypes = {
  avatar: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  date: PropTypes.string,
  className: PropTypes.string,
  onClickFollow: PropTypes.func,
  isFollowed: PropTypes.bool,
  href: PropTypes.string
};

StoryHead.defaultProps = {
  href: "#"
};

export default StoryHead;
