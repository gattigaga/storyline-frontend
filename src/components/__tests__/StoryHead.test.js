import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import StoryHead, { MiniButton } from "../StoryHead";

describe("StoryHead", () => {
  const setup = propOverrides => {
    const props = {
      avatar: require("../../../assets/images/dummy.png"),
      name: "Brendan Eich",
      onClickFollow: jest.fn(),
      date: "Jan 01, 2018",
      ...propOverrides
    };

    const wrapper = shallow(<StoryHead {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders default", () => {
    const { wrapper } = setup({ isFollowed: true });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should calls 'onClick' callback while follow button pressed", () => {
    const { wrapper, props } = setup();

    wrapper.find(MiniButton).simulate("click");

    expect(props.onClickFollow).toBeCalled();
  });
});
