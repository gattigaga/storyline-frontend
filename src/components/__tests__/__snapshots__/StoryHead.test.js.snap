// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryHead should renders default 1`] = `
<styled.div>
  <Link
    replace={false}
    to="#"
  >
    <Avatar
      size={64}
      src="dummy.png"
    />
  </Link>
  <styled.div>
    <styled.div>
      <Styled(Link)
        to="#"
      >
        Brendan Eich
      </Styled(Link)>
      <Styled(Button)
        caption="Follow"
        isOutlined={true}
        onClick={[Function]}
      />
    </styled.div>
    <styled.p />
    <styled.p>
      Jan 01, 2018
    </styled.p>
  </styled.div>
</styled.div>
`;

exports[`StoryHead should renders default 2`] = `
<styled.div>
  <Link
    replace={false}
    to="#"
  >
    <Avatar
      size={64}
      src="dummy.png"
    />
  </Link>
  <styled.div>
    <styled.div>
      <Styled(Link)
        to="#"
      >
        Brendan Eich
      </Styled(Link)>
      <Styled(Button)
        caption="Unfollow"
        isOutlined={false}
        onClick={[Function]}
      />
    </styled.div>
    <styled.p />
    <styled.p>
      Jan 01, 2018
    </styled.p>
  </styled.div>
</styled.div>
`;
